Index: novaprova-1.1/Makefile
===================================================================
--- novaprova-1.1.orig/Makefile	2013-01-16 00:42:37.000000000 +1100
+++ novaprova-1.1/Makefile	2013-01-25 20:50:17.000000000 +1100
@@ -22,15 +22,13 @@
 libdir=		$(exec_prefix)/lib
 mandir=		$(prefix)/share/man
 pkgdocdir=	$(prefix)/share/doc/$(PACKAGE)
-configdir=	$(libdir)/pkgconfig
+pkgconfigdir=	$(libdir)/pkgconfig
 
 CC=		g++
 CXX=		g++
 CDEBUGFLAGS=	-g
 COPTFLAGS=	-O0
-CDEFINES=	-I. \
-		$(shell ./platform.sh --cflags) \
-		$(shell pkg-config --cflags libxml++-2.6)
+CDEFINES=	-I. $(platform_CFLAGS) $(libxml_CFLAGS)
 CWARNFLAGS=	-Wall -Wextra
 CFLAGS=		$(CDEBUGFLAGS) $(COPTFLAGS) $(CWARNFLAGS) $(CDEFINES)
 CXXFLAGS=	$(CFLAGS)
@@ -38,8 +36,8 @@
 RANLIB=		ranlib
 depdir=		.deps
 
-SUBDIRS_PRE=
-SUBDIRS_POST=	tests doc/get-start
+SUBDIRS_PRE=	doc/get-start
+SUBDIRS_POST=	tests
 
 all clean distclean check install docs:
 	@for dir in $(SUBDIRS_PRE) ; do $(MAKE) -C $$dir $@ ; done
@@ -48,12 +46,21 @@
 
 install check: all
 
-all-local: .config-ok libnovaprova.a novaprova.pc
+all-local: libnovaprova.a novaprova.pc
 
-.config-ok:
+include .config
+
+.config:
 	@./platform.sh --verify
-	@pkg-config --cflags libxml++-2.6 > /dev/null
-	@touch $@
+	@echo "platform_CFLAGS="`./platform.sh --cflags` >> $@
+	@got_libxml=no ; for p in libxml++-2.6 libxml++ ; do \
+	    if pkg-config --exists $$p ; then \
+		echo "libxml=$$p" >> $@ ;\
+		echo "libxml_CFLAGS="`pkg-config --cflags $$p` >> $@ ;\
+		got_libxml=yes ;\
+	    fi ;\
+	done && [ $$got_libxml = yes ]
+	@echo ==== ; cat $@ ; echo ====
 
 libnovaprova_SOURCE= \
 		np.c \
@@ -137,15 +144,12 @@
 -include $(libnovaprova_DFILES)
 
 %.o: %.cxx
-	$(COMPILE.C) -o $@ $<
+	@mkdir -p $(dir $(depdir)/$(patsubst %.cxx,%.d,$<))
+	$(COMPILE.C) -MD -MM -MF $(depdir)/$(patsubst %.cxx,%.d,$<) -MT $(patsubst %.cxx,%.o,$<) -o $@ $<
 
-$(depdir)/%.d: %.cxx
-	@[ -d $(@D) ] || mkdir -p $(@D)
-	$(COMPILE.C) -MM -MF $@ -MT $(patsubst %.cxx,%.o,$<) $<
-
-$(depdir)/%.d: %.c
-	@[ -d $(@D) ] || mkdir -p $(@D)
-	$(COMPILE.c) -MM -MF $@ -MT $(patsubst %.c,%.o,$<) $<
+%.o: %.c
+	@mkdir -p $(dir $(depdir)/$(patsubst %.c,%.d,$<))
+	$(COMPILE.c) -MD -MM -MF $(depdir)/$(patsubst %.c,%.d,$<) -MT $(patsubst %.c,%.o,$<) -o $@ $<
 
 libnovaprova.a: $(libnovaprova_OBJS)
 	$(AR) $(ARFLAGS) libnovaprova.a $(libnovaprova_OBJS)
@@ -165,11 +169,12 @@
 docs-local: Doxyfile
 	$(RM) -r doc/api-ref doc/man
 	doxygen
+	$(RM) doc-$(VERSION)
 	ln -s doc doc-$(VERSION)
 	tar -chjvf doc-$(VERSION).tar.bz2 $(addprefix doc-$(VERSION)/,$(DOC_DELIVERABLES))
 	$(RM) doc-$(VERSION)
 
-install: docs
+install:
 
 install-local:
 	$(INSTALL) -d $(DESTDIR)$(includedir)/novaprova/np
@@ -183,7 +188,8 @@
 	$(INSTALL) doc/man/man3/*.3 $(DESTDIR)$(mandir)/man3
 	$(INSTALL) -d $(DESTDIR)$(pkgdocdir)/api-ref
 	$(INSTALL) doc/api-ref/* $(DESTDIR)$(pkgdocdir)/api-ref
-	$(INSTALL) novaprova.pc $(DESTDIR)$(configdir)
+	$(INSTALL) -d $(DESTDIR)$(pkgconfigdir)
+	$(INSTALL) novaprova.pc $(DESTDIR)$(pkgconfigdir)
 
 clean-local:
 	$(RM) libnovaprova.a $(libnovaprova_OBJS)
@@ -192,6 +198,7 @@
 	$(RM) -r doc/api-ref doc/man
 	$(RM) -r $(depdir)
 	$(RM) novaproca.pc
+	$(RM) .config
 
 check-local:
 
@@ -205,8 +212,9 @@
 	    -e 's|@libdir@|$(libdir)|' \
 	    -e 's|@mandir@|$(mandir)|' \
 	    -e 's|@pkgdocdir@|$(pkgdocdir)|' \
-	    -e 's|@configdir@|$(configdir)|' \
+	    -e 's|@pkgconfigdir@|$(pkgconfigdir)|' \
 	    -e 's|@mandir@|$(mandir)|' \
+	    -e 's|@libxml@|$(libxml)|' \
 	    < $< > $@
 
 DIST_TARBALL=	$(PACKAGE)-$(VERSION).tar.bz2
Index: novaprova-1.1/novaprova.pc.in
===================================================================
--- novaprova-1.1.orig/novaprova.pc.in	2012-08-15 09:28:44.000000000 +1000
+++ novaprova-1.1/novaprova.pc.in	2013-01-25 20:49:01.000000000 +1100
@@ -6,6 +6,6 @@
 Name: NovaProva
 Description: New generation unit test framework for C
 Version: @VERSION@
-Requires: libxml++-2.6
+Requires: @libxml@
 Libs: -L${libdir} -lnovaprova -lstdc++ -lbfd -ldl -lrt
 Cflags: -I${includedir}
